app/models/user.rb

```diff
@@ -73,6 +73,13 @@ def any_admin?
     admin? || beach_admin?
   end
 
+  def reporter_rank
+    my_rank = StatusCrowdsourceUser.select('COUNT(*) AS count, user_id, RANK() over (ORDER BY COUNT(*) DESC)')
+      .group(:user_id)
+      .where(user_id: id)
+    my_rank.to_a&.first&.rank || 0
```

Comment:
Would this work?

```
StatusCrowdsourceUser.select('COUNT(*) AS count, user_id, RANK() over (ORDER BY COUNT(*) DESC)')
      .group(:user_id)
      .where(user_id: id)
      .first&.rank || 0
```

Unsure why you're `to_a&.`ing to get the first element. Does `group by` break `#first`?

Reply:
Yes, `#first` was throwing an error regarding the inclusion of `id` on the select and group by clause. I have since changed this, because filtering by user_id was giving me always the first position. So I first select the first 100 and check if the user is present in those.